lib/index.js

```diff
@@ -103,5 +114,32 @@ function transform (post) {
   return post
 }
 
+function writeTemplate (ct, compiler, compilation, addDataTo) {
+  const data = addDataTo.rooftop[ct.name]
+  const filePath = path.join(compiler.options.context, ct.template.path)
+
+  return node.call(fs.readFile.bind(fs), filePath, 'utf8')
+    .then((template) => {
+      return data.map((item) => {
+        addDataTo = Object.assign(addDataTo, { item: item })
+        compiler.request = filePath
+        let plugins = compiler.options.posthtml
+        if (typeof plugins === 'function') plugins = plugins.call(this, compiler)
+        if (typeof plugins === 'object') plugins = plugins.defaults
+
+        return posthtml(plugins)
+          .process(template)
+          .then((r) => r.html)
+          .then((rendered) => {
+            compilation.assets[ct.template.output(item)] = {
+              source: () => rendered,
+              size: () => rendered.length
+            }
+          })
+      })
+    })
+}
+
 module.exports = Rooftop
 module.exports.transform = transform
+module.exports.writeTemplate = writeTemplate
```

Comment:
Left that in by mistake. I put that in earlier while testing something and forgot to take it out. Will remove now.